athenzdomains.go

```diff
@@ -0,0 +1,154 @@
+package webhook
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	v1 "github.com/yahoo/k8s-athenz-istio-auth/pkg/apis/athenz/v1"
+	athenzClientset "github.com/yahoo/k8s-athenz-istio-auth/pkg/client/clientset/versioned"
+	athenzInformer "github.com/yahoo/k8s-athenz-istio-auth/pkg/client/informers/externalversions/athenz/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	replacer = strings.NewReplacer(".*", ".*", "*", ".*")
+)
+
+// CRMap - cr cache
+type CRMap struct {
+	RoleToPrincipals map[string][]*zms.RoleMember
+	RoleToAssertion  map[string][]SimpleAssertion
+}
+
+// SimpleAssertion - processed policy
+type SimpleAssertion struct {
+	resource *regexp.Regexp
+	action   *regexp.Regexp
+	effect   string
+}
+
+// Cache - cache for athenzdomains CR
+type Cache struct {
+	CrIndexInformer cache.SharedIndexInformer
+	DomainMap       *map[string]CRMap
+}
+
+// BuildCache - generate new athenzdomains cr cache
+func BuildCache() (*Cache, error) {
```

Comment:
we should probably create the informer and pass it in Config

Reply:
will it make sense to create a separate `zpeclient.go` which has the informer event handlers, cache for the athenz domains, and a `authorize(ctx, principal, check)` function that evaluates using the local cache, and the api would probably call zpeclient.authorize() and then compare the result with client.authorize()?